library/time-library/src/main/java/uk/gov/gchq/gaffer/time/LongTimeSeries.java

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright 2017 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.gaffer.time;
+
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+
+import uk.gov.gchq.gaffer.commonutil.CommonTimeUtil;
+import uk.gov.gchq.gaffer.commonutil.CommonTimeUtil.TimeBucket;
+import uk.gov.gchq.gaffer.commonutil.ToStringBuilder;
+
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.SortedMap;
+import java.util.SortedSet;
+import java.util.TreeMap;
+import java.util.TreeSet;
+
+/**
+ * This is a time series where the values are {@link Long}s. When the time series
+ * is created, a {@link TimeBucket} is specified. When timestamps are added, they
+ * are rounded to the nearest bucket, e.g. if a {@link TimeBucket} of
+ * <code>MINUTE</code> is specified, and a timestamp of January 1st 2017, 12:34:56
+ * is added then the seconds are removed so that the value is associated to
+ * 12:34.
+ */
+public class LongTimeSeries implements TimeSeries<Long> {
```

Comment:
This class doesn't look like it will serialise to/from JSON properly. You have 2 getters for returning the time series information: getInstants and getTimeSeries, one of these needs to be annotated with @JsonIgnore otherwise you will get the same data twice in JSON. You also need a default constructor and a setter for the timeSeries and timeBucket fields. 

As a lot of Gaffer interactions are done via REST in JSON, it would be great if this class worked over REST. 

Your LongTimeSeriesTest should then extend JSONSerialisationTest and that will add a json serialisation test.